### PR TITLE
Adding flag for disabling the friendly error system

### PR DESCRIPTION
**Pull request by mikewesthad**
**Summary from #1512, #1329, #971:**  

The friendly error logging system causes a significant slowdown (e.g. 10x slowdown in Chrome for simple functions like `random` or `sin`).  

In global instance mode with unminified p5, p5 properties are added to the global scope using getters/setters via `Object.defineProperty` in [core.js#L593](https://github.com/processing/p5.js/blob/master/src/core/core.js#L593).  This makes it possible to log an error if a beginner (or not-so-beginner) inadvertently overrides a p5 property (e.g. `text`). This is a really helpful feature, but comes with a hefty performance cost.

**Proposed solution:**

Add a property to the p5 constructor function `disableFriendlyErrors` which can be used to prevent the friendly error getters/setters.  That way beginners get the friendly errors by default, but adding `p5.disableFriendlyErrors = true` to the top of a sketch will boost performance.  (I attached this property to the p5 constructor over p5.prototype - seems to make more sense that way.)

**Some Results:**

tl;dr: Huge gains in Chrome, reasonable gains in p5 editor & FF, no real gains in IE/Edge. I'm opening this PR as a solution to part of #1512, but it's disappointing that it's not a magic bullet for the p5 editor slowdown.  (Maybe that's because the chromium engine powering the editor is >1 year old?)

```
Calling p5 methods in global mode 10000000x times. 

Chrome, Windows:

Friendly Errors:  enabled       ->       disabled
random:           1518.62ms     ->       320.28md
abs:              1210.32ms     ->       91.99ms
sin:              1712.18ms     ->       478.58ms

Same thing in p5 Editor (built against this PR's p5.js):

Friendly Errors:  enabled       ->       disabled     
random:           3077.91ms     ->       1798.99ms
abs:              3153.38ms     ->       21935.76ms
sin:              3579.56ms     ->       2560.79ms


Times vary in FF but performance boost ranges from 2x-10x.
IE11/Edge minimal to zero gains.
```

Could use a better test (#1524), but for now, something manual:

```js
p5.disableFriendlyErrors = true;

var iterations = 10000000;

function setup() {
  console.log("Friendly Errors: " +
    (p5.disableFriendlyErrors) ? "disabled" : "enabled");

  var start = performance.now();
  for (var i = 0; i < iterations; i += 1) {
    random();
  }
  var elapsed = performance.now() - start;
  console.log("random: " + elapsed.toFixed(2) + "ms");

  var start = performance.now();
  for (var i = 0; i < iterations; i += 1) {
    abs(i);
  }
  var elapsed = performance.now() - start;
  console.log("abs: " + elapsed.toFixed(2) + "ms");

  var start = performance.now();
  for (var i = 0; i < iterations; i += 1) {
    sin(i/iterations * (15 * 3.14));
  }
  var elapsed = performance.now() - start;
  console.log("sin: " + elapsed.toFixed(2) + "ms");
}
```

### CODE REVIEW

**Comment by lmccart** [comment]
this is great! my one question is whether is makes more sense to have: 

```javascript
p5.disableFriendlyErrors = true;

function setup() {
  createCanvas(100, 100);
  // etc
}
```

or a function call within setup, something like:
```javascript
function setup() {
  friendlyErrors(false); // or disableFriendlyErrors();
  createCanvas(100, 100);
}
```

maybe the second is a little more in line with the flow people use to think about and write p5, ie it is similar to the way you would [disable pixel density scaling](http://p5js.org/reference/#/p5/pixelDensity). or, do you feel this is on a meta level above the code and should be outside of setup?

**Comment by mikewesthad** [comment]
Yeah, I do agree that having it be a method on p5.prototype would be more processing-like.  

The reason why I put it on the p5 constructor is because the value needs to be set before `setup` is called. Basically, the value needs to set before the page triggers a load event.  That's when the global p5 instance is created and the getters/setters are put in place.

I figured this would be cleaner than providing a way to rebind all the p5 properties/methods to the window on-the-fly via `friendlyErrors(false)`.  I was worried that rebinding might cause some weird edge cases, but let me take a look and see.

**Comment by lmccart** [comment]
ah, I was thinking that might be the case. ok let's stick with this then.
